src/code/z_actor.c

```diff
@@ -1057,20 +1057,37 @@ void func_8002DF18(PlayState* play, Player* player) {
     func_8006DC68(play, player);
 }
 
-s32 func_8002DF38(PlayState* play, Actor* actor, u8 csAction) {
+/**
+ * Sets a Player Cutscene Action specified by `csAction`.
+ * There are no safety checks to see if Player is already in some form of a cutscene state.
+ * This will instantly take effect.
+ *
+ * `haltActorsDuringCsAction` being set to false in this function means that all actors will
+ * be able to update while Player is performing the cutscene action.
+ */
+s32 Player_SetCsAction(PlayState* play, Actor* csActor, u8 csAction) {
     Player* player = GET_PLAYER(play);
 
     player->csAction = csAction;
-    player->unk_448 = actor;
+    player->csActor = csActor;
     player->cv.haltActorsDuringCsAction = false;
 
     return true;
 }
 
-s32 func_8002DF54(PlayState* play, Actor* actor, u8 csAction) {
+/**
+ * Sets a Player Cutscene Action specified by `csAction`.
+ * There are no safety checks to see if Player is already in some form of a cutscene state.
+ * This will instantly take effect.
+ *
+ * `haltActorsDuringCsAction` being set to true in this function means that eventually `PLAYER_STATE1_29` will be set.
+ * This makes it so actors belonging to categories `ACTORCAT_ENEMY` and `ACTORCAT_MISC` will not update
+ * while Player is in the `csAction` state.
```

Comment:
Is `haltActorsDuringCsAction` reset at the end of the csAction (if they even end?),
or do you mean that csActions are only (meant to be) set by `Player_SetCsAction.*`, and so `haltActorsDuringCsAction` would always be set explicitly at the start of a csAction instead of ever implicitly retaining a previous value?

(can try to rephrase if this isn't clear lol)

Reply:
`haltActorsDuringCsAction` is never reset in any case (even when used as a bgcam index for sliding doors)

Cs Actions are only meant to be started with `Player_SetCsAction`, `Player_SetCsActionWithHaltedActors` , or `play->tryPlayerCsAction`

`haltActorsDuringCsAction` is only relevant if a csAction is started through the z_actor functions, not the "try" play state playback. as far as I can tell the playstate playback will not halt actors, but it never checks `haltActorsDuringCsAction` anyway

Reply:
Discussed (this and more) on discord from https://discord.com/channels/688807550715560050/688851317593997489/1160630768092053515 to https://discord.com/channels/688807550715560050/688851317593997489/1160659982501552199


We left it at adding a note about jank `haltActorsDuringCsAction` if someone wanted to toggle the behavior 

> note: due to how player implements initializing the cutscene action state, `haltActorsDuringCsAction` will only be considered the first time player starts a csAction. Player must leave the cutscene action state before halting actors can be toggled.

I'm personally still hesitant on the difference between
"csAction"="a csAction value handled by `Player_Action_CsAction`"
vs "cutscene action state"="the current player action function being `Player_Action_CsAction`"
being clear enough / not confusing though :/

I think this stems from `Player_Action_CsAction` having "action" twice but idk how to address it

Reply:
Its always gonna have action twice in the name cause of how it works. Its the action function that runs cutscene actions. This was kinda by design when I decided to name cs modes as cs actions.

Im not sure the name of that function really plays into the confusion. The comment just needs to adequately describe the fact that player is in a cs action state. Other words like "cutscene action playback state" can be used. or can bring back "mode" to describe the overall result of player state, idk.

This detail in general is pretty minor and I dont think its worth getting too hung up on personally, but more opinions welcome

Reply:
I agree this detail is not a big deal

could you just change to
```suggestion
 * while Player is in the cutscene action state.
```
so it at least vaguely sounds like it's "as long as player uses the cutscene action action" and not for the duration of the specific `csAction`

same with "`while Player is performing the cutscene action`" on the sister function